sources/packages/api/src/database/entities/institution.model.ts

```diff
@@ -0,0 +1,88 @@
+import {
+  Column,
+  Entity,
+  ManyToMany,
+  PrimaryGeneratedColumn,
+  JoinTable,
+} from "typeorm";
+import { TableNames } from "../constant";
+import { RecordDataModel } from "./record.model";
+import {
+  InstitutionAddress,
+  InstitutionPrimaryContact,
+  LegalAuthorityContact,
+} from "../../types";
+import { User } from "./user.model";
+
+@Entity({ name: TableNames.Institution })
+export class Institution extends RecordDataModel {
+  @PrimaryGeneratedColumn()
+  id: number;
+
+  @Column({
+    name: "guid",
+  })
+  guid: string;
+
+  @Column({
+    name: "legal_operating_name",
+  })
+  legalOperatingName: string;
+
+  @Column({
+    name: "operating_name",
+    nullable: true,
+  })
+  operatingName: string;
+
+  @Column({
+    name: "primary_phone",
+  })
+  primaryPhone: string;
+
+  @Column({
+    name: "primary_email",
+  })
+  primaryEmail: string;
+
+  @Column({
+    name: "website",
+  })
+  website: string;
+  @Column({
+    name: "regulating_body",
+  })
+  regulatingBody: string;
+
+  @Column({
+    name: "established_date",
+  })
+  establishedDate: Date;
+
+  @Column({
+    name: "primary_contact",
+    type: "jsonb",
+    nullable: false,
+  })
+  institutionPrimaryContact: InstitutionPrimaryContact;
+
+  @Column({
+    name: "legal_authority_contact",
+    type: "jsonb",
+  })
+  legalAuthorityContact: LegalAuthorityContact;
+
+  @Column({
+    name: "institution_address",
+    type: "jsonb",
+  })
+  institutionAddress: InstitutionAddress;
+
+  @ManyToMany(() => User, { eager: true, cascade: true })
```

Comment:
I would avoid adding the "eager: true".
I believe that when loading an institution we should request it if needed.

Reply:
Is there any case for our application to eager:true?

Reply:
None that I can see how now. My understanding of this "eager: true" is that every time that we load an institution record all users-related records will be loaded. In general, I would avoid loading from DB more data that we need.

Reply:
I agree with @andrewsignori-aot Eager will additional overhead on institution data fetch operation.

Reply:
Done